### ISSUE TITLE
can't show the whole files if torrent cantains hundreds of files

### ISSUE THREAD

**Issue by i990049** [elgatito/plugin.video.elementum#1086]
elementum0.1.106
only show two m2ts files on my kodi of Android TV BOX.
[test.zip](https://github.com/user-attachments/files/17776338/test.zip)

**Comment by antonsoroko** [comment]
@i990049 
1.
about 
> only show two m2ts files on my kodi of Android TV BOX.

that's because this MEGA torrent has not only regular video files (mkv, mp4, avi), but it also has 2 Bluray's folders (path has `BDMV/STREAM`).
so elementum treat this torrent as special bluray torrent. so you see only bluray things.

```
warning <general>: [plugin.video.elementum] INFO  bittorrent   ▶ GetCandidateFiles  Found Blu-Ray: 动画片/[冰雪奇缘].Frozen.2013.Blu-ray.x264.DTS.4Audios.MiniBD1080P-CMCT/BDMV/STREAM/00000.m2ts
warning <general>: [plugin.video.elementum] INFO  bittorrent   ▶ GetCandidateFiles  Found Blu-Ray: 动画片/[神偷奶爸3]Despicable.Me.3.2017.BluRay.REMUX.1080p.AVC.DTS-HD.MA7.1-HDS/BDMV/STREAM/00000.m2ts
```


i am not sure that @elgatito  will agree to add some hack to this logic to show mix of regular files (mkv, mp4) and bluray files. i also not sure that this is useful. this is the first time i see such torrent.

responsible code https://github.com/elgatito/elementum/blob/638d5c307b53763e752a698252ecb03de06e085a/bittorrent/torrent.go#L1792

2.
about 
> It will be convenient if there is file name search filter when open a magnet link or torrent.

since usually torrent has not many files, i do not know if this is really useful.

anyway, someone who is good in developing kodi windows/dialogs should design such custom window/dialog. 
so i would say - unless you will implement this feature yourself - i would not expect this feature.

**Comment by antonsoroko** [comment]
to answer

>how to solve the previous torrent with hundred files but elementum only display two files？

from https://github.com/elgatito/plugin.video.elementum/issues/1147#issuecomment-3245889957


@i990049 only by changing code, as it was already answered here.

see https://github.com/elgatito/plugin.video.elementum/issues/1086#issuecomment-2479831525
and https://github.com/elgatito/plugin.video.elementum/issues/1086#issuecomment-2690887044

but nobody from developers needs such feature since it is very extremely rare situation. i mean that "a pile of different types of videos in one torrent" situation that you showed. elementum is designed for "normal" video torrents.

if you need to download "dumpster" torrent - it is better to use regular torrent client. for android tv - personally I use BiglyBT.

but if you want to learn golang and then implement such feature it by yourself - then cool, otherwise - i would not expect that someone will do this for you.

**BTW,** you can edit that huge torrent file and remove all Blu-Ray folders from it - thus all other files will be visible again in elementum.

that's all ideas i have at the moment.

**Comment by i990049** [comment]
bitlord Android APP is better than bigly bt APP while playing unfinished download file but without any subtitle and not support by Nexusphp. Can elementum edit file type remove all Blu-Ray folders through Filename Extension such as m2ts？



---Original---
From: "Anton ***@***.***&gt;
Date: Wed, Sep 3, 2025 02:46 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [elgatito/plugin.video.elementum] can't show the whole files iftorrent cantains hundreds of files (Issue #1086)


antonsoroko left a comment (elgatito/plugin.video.elementum#1086)
 
to answer
  
how to solve the previous torrent with hundred files but elementum only display two files？
  
from #1147 (comment)
 
@i990049 only by changing code, as it was already answered here.
 
see #1086 (comment)
 and #1086 (comment)
 
but nobody from developers needs such feature since it is very extremely rare situation. i mean that "a pile of different types of videos in one torrent" situation that you showed. elementum is designed for "normal" video torrents.
 
if you need to download "dumpster" torrent - it is better to use regular torrent client. for android tv - personally I use BiglyBT.
 
but if you want to learn golang and then implement such feature it by yourself - then cool, otherwise - i would not expect that someone will do this for you.
 
BTW, you can edit that huge torrent file and remove all Blu-Ray folders from it - thus all other files will be visible again in elementum.
 
that's all ideas i have at the moment.

**Comment by antonsoroko** [comment]
@i990049 

>Can elementum edit file type remove all Blu-Ray folders through Filename Extension such as m2ts？

no. elementum does not have such functionality because it is very-very specific functionality.

i mean someone can write a code that will add a setting to exclude some files types from torrent.

but then i guess it is better to add support of mixed files (e.g. bluray files + .mkv files) - but this will be much more complex to implement.

i want to be honest: personally i do not see value in implement such features since i have never used such torrent files. thus personally i do not want to implement such very-very specific features.

**Comment by antonsoroko** [comment]
@i990049 about
>bitlord Android APP is better than bigly

Maybe for android (phone version), but I was talking about "android tv" (special android version for tv and and TV box).
Biglybt is available in Google play for Android TV and it has UI optimized for tv remote. That's why I use it, if I need torrent client on my Android TV box.

(I know there are many "custom" android aosp builds for tv boxes, but those are not official android tv.)

Anyway, what I tried to say is that to download "dumpster" torrent - it is better to use regular torrent client.

**Comment by antonsoroko** [comment]
@i990049 about
>bitlord Android APP is better than bigly

Maybe for android (phone version), but I was talking about "android tv" (special android version for tv and and TV box).
Biglybt is available in Google play for Android TV and it has UI optimized for tv remote. That's why I use it, if I need torrent client on my Android TV box.

(I know there are many "custom" android aosp builds for tv boxes, but those are not official android tv.)

Anyway, what I tried to say is that to download "dumpster" torrent - it is better to use regular torrent client.

Btw, you can try to use "torrserver" for such strange torrents.
